Source/Core/Core/TitleDatabase.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <functional>
+#include <iostream>
+
+#include "Core/TitleDatabase.h"
+
+#include "Common/FileUtil.h"
+#include "Common/MsgHandler.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "DiscIO/Enums.h"
+
+namespace Core
+{
+static std::string GetLanguageCode(DiscIO::Language language)
+{
+  switch (language)
+  {
+  case DiscIO::Language::LANGUAGE_JAPANESE:
+    return "ja";
+  case DiscIO::Language::LANGUAGE_ENGLISH:
+    return "en";
+  case DiscIO::Language::LANGUAGE_GERMAN:
+    return "de";
+  case DiscIO::Language::LANGUAGE_FRENCH:
+    return "fr";
+  case DiscIO::Language::LANGUAGE_SPANISH:
+    return "es";
+  case DiscIO::Language::LANGUAGE_ITALIAN:
+    return "it";
+  case DiscIO::Language::LANGUAGE_DUTCH:
+    return "nl";
+  case DiscIO::Language::LANGUAGE_SIMPLIFIED_CHINESE:
+    return "zh_CN";
+  case DiscIO::Language::LANGUAGE_TRADITIONAL_CHINESE:
+    return "zh_TW";
+  case DiscIO::Language::LANGUAGE_KOREAN:
+    return "ko";
+  default:
+    return "en";
+  }
+}
+
+using Map = std::unordered_map<std::string, std::string>;
+
+static bool LoadMap(const std::string& file_path, Map& map,
+                    std::function<bool(const std::string& game_id)> predicate)
+{
+  std::ifstream txt;
+  OpenFStream(txt, file_path, std::ios::in);
+
+  if (!txt.is_open())
+    return false;
+
+  std::string line;
+  while (std::getline(txt, line))
+  {
+    if (line.empty())
+      continue;
+
+    const size_t equals_index = line.find('=');
+    if (equals_index != std::string::npos)
+    {
+      const std::string game_id = StripSpaces(line.substr(0, equals_index));
+      if (predicate(game_id))
+        map[game_id] = StripSpaces(line.substr(equals_index + 1));
+    }
+  }
+  return true;
+}
+
+static bool IsGCTitle(const std::string& game_id)
```

Comment:
Please add a comment that this function only is supposed to be used with wiitdb.txt format game IDs, not Dolphin format game IDs. (TurboGrafx-16 titles will be misdetected if a Dolphin format game ID is passed in.)